x-pack/plugins/security_solution/public/common/components/filter_group/__tests__/filter_group.test.tsx

```diff
@@ -0,0 +1,670 @@
+      render(<TestComponent />);
+
+      updateControlGroupInputMock(initialInputData as ControlGroupInput);
+
+      await openContextMenu();
+
+      fireEvent.click(screen.getByTestId(TEST_IDS.CONTEXT_MENU.EDIT));
+
+      // delete some panels
+      const newInputData = {
+        ...initialInputData,
+        panels: {
+          '0': initialInputData.panels['0'],
+        },
+      } as ControlGroupInput;
+
+      updateControlGroupInputMock(newInputData);
+
+      await waitFor(() => {
+        expect(screen.getByTestId(TEST_IDS.SAVE_CHANGE_POPOVER)).toBeVisible();
+      });
+
+      fireEvent.mouseOver(screen.getByTestId(TEST_IDS.SAVE_CONTROL));
+      fireEvent.mouseOut(screen.getByTestId(TEST_IDS.SAVE_CONTROL));
+      await waitFor(() => {
+        expect(screen.queryByTestId(TEST_IDS.SAVE_CHANGE_POPOVER)).toBeNull();
+      });
+
+      fireEvent.mouseOver(screen.getByTestId(TEST_IDS.SAVE_CONTROL));
+      await waitFor(() => {
+        expect(screen.queryByTestId(TEST_IDS.SAVE_CHANGE_POPOVER)).toBeVisible();
+      });
+    });
+  });
+
+  describe('Filter Changed Banner', () => {
+    beforeAll(() => {
+      (useGetInitialUrlParamValue as jest.Mock).mockImplementation(() => {
+        return () => [
+          {
+            fieldName: 'abc',
+          },
+        ];
+      });
+    });
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      global.localStorage.clear();
+    });
+
+    it('should show banner if url filter and stored filters are not same', async () => {
+      global.localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(initialInputData));
+
+      render(<TestComponent />);
+      updateControlGroupInputMock(initialInputData as ControlGroupInput);
+      await waitFor(() => {
+        expect(screen.getByTestId(TEST_IDS.FILTERS_CHANGED_BANNER)).toBeVisible();
+      });
+    });
+
+    it('should use url filters if url and stored filters are not same', async () => {
+      global.localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(initialInputData));
+      render(<TestComponent />);
+      updateControlGroupInputMock(initialInputData as ControlGroupInput);
+      expect(controlGroupMock.addOptionsListControl.mock.calls.length).toBe(2);
+      expect(controlGroupMock.addOptionsListControl.mock.calls[0][1]).toMatchObject({
+        hideExclude: true,
+        hideSort: true,
+        hidePanelTitles: true,
+        placeholder: '',
+        ...DEFAULT_DETECTION_PAGE_FILTERS[0],
+      });
+
+      expect(controlGroupMock.addOptionsListControl.mock.calls[1][1]).toMatchObject({
+        hideExclude: true,
+        hideSort: true,
+        hidePanelTitles: true,
+        placeholder: '',
+        fieldName: 'abc',
+      });
+
+      await waitFor(() => {
+        expect(screen.getByTestId(TEST_IDS.FILTERS_CHANGED_BANNER)).toBeVisible();
+      });
+    });
+
+    it('should ignore url params if there is an error in using them', async () => {
```

Comment:
I don't know if this matters but... If you skip this test, the "Restore from local storage" group fails. How are they related? Worried this test isn't cleaning up something

Reply:
Great catch.. there was issue of `mockImplementation` vs `mockImplementationOnce`. Thanks for your thorough review.